pipeline/transformer/transformer_marc_common/src/main/scala/weco/pipeline/transformer/marc_common/transformers/MarcProduction.scala

```diff
@@ -0,0 +1,309 @@
+  //
+  // The rules are as follows:
+  //
+  //  - Populate "places" from subfield "a" and type as "Place"
+  //  - Populate "agents" from subfield "b" and type as "Agent"
+  //  - Populate "dates" from subfield "c" and type as "Period"
+  //
+  // The production function is set based on the second indicator, as defined
+  // in the MARC spec.
+  //
+  //  - 0 = Production
+  //  - 1 = Publication
+  //  - 2 = Distribution
+  //  - 3 = Manufacture
+  //
+  // The MARC spec specifies another value for the production function:
+  //
+  //  - 4 = Copyright notice date
+  //
+  // We'll be putting copyright information in a separate part of the domain
+  // model, so we drop any fields with indicator 4 for production events.
+  //
+  // Note that a, b and c are repeatable fields.
+  //
+  // https://www.loc.gov/marc/bibliographic/bd264.html
+  //
+  private def getProductionFrom264Fields(
+    record: MarcRecord
+  ) =
+    record
+      .fieldsWithTags("264")
+      .filterNot {
+        field =>
+          field.indicator2.contains("4") || field.indicator2.contains(" ")
+      }
+      .map {
+        field =>
+          val label = labelFromSubFields(field)
+          val places = placesFromSubfields(field, subfieldTag = "a")
+          val agents = agentsFromSubfields(field, subfieldTag = "b")
+          val dates = datesFromSubfields(field, subfieldTag = "c")
+
+          val productionFunctionLabel = field.indicator2 match {
+            case "0" => "Production"
+            case "1" => "Publication"
+            case "2" => "Distribution"
+            case "3" => "Manufacture"
+            case other =>
+              throw CataloguingException(
+                record = record,
+                message =
+                  s"Unrecognised second indicator for production function: [$other]"
+              )
+          }
+
+          val productionFunction =
+            Some(Concept(label = productionFunctionLabel))
+
+          ProductionEvent(
+            label = label,
+            places = places,
+            agents = agents,
+            dates = dates,
+            function = productionFunction
+          )
+      }
+
+  private def marc264OnlyContainsCopyright(
+    marcFields: List[MarcField]
+  ): Boolean =
+    marcFields match {
+      case List(
+            MarcField("264", Seq(MarcSubfield("c", content)), _, _, _)
+          ) =>
+        content.matches("^©\\d{4}$")
+      case _ => false
+    }
+
+  private def marc264IsOnlyPunctuation(
+    marc264fields: List[MarcField]
+  ): Boolean =
+    marc264fields
+      .map { _.subfields.map(_.content).mkString("") }
+      .forall { _ matches "^[:,]*$" }
+
+  /** Populate the production data if both 260 and 264 are present.
+    *
+    * In general, this is a cataloguing error, but sometimes we can do something
+    * more sensible depending on if/how they're duplicated.
+    */
+  private def getProductionFromBothFields(record: MarcRecord) = {
```

Comment:
I know this is just a straight copy from its previous location, but having read it just now, I fancy it is ripe for a refactor at some point in the future.

1. all the (successful) branches return the same thing, (i.e. the 260) so it might be preferable to OR all these conditions, possibly in a separate "shouldDiscard264" function
2. Because they only pull 260, it's not really get from both,
3. this is only called in a place where we already have the output, this is really just choosing whether to keep the output or throw an exception, but it's regenerating the output

Reply:
I will have a go at this in a fresh PR.